src/sentry/integrations/jira/integration.py

```diff
@@ -958,6 +961,48 @@ def get_resolve_sync_action(self, data: Mapping[str, Any]) -> ResolveSyncAction:
             should_unresolve=c_from in done_statuses and c_to not in done_statuses,
         )
 
+    def migrate_issues(self):
+        for project in Project.objects.filter(organization_id=self.organization_id):
+            plugin = None
+            for p in plugins.for_project(project):
+                if self.model.provider.startswith(p.slug) and p.get_option(
+                    "default_project", project
+                ):
+                    plugin = p
+                    break
+
+            if not plugin:
+                continue
+            if plugin.get_option("instance_url", project).rstrip("/") != self.model.metadata.get(
+                "base_url"
+            ).rstrip("/"):
+                continue
```

Comment:
Maybe this is kinda self explanatory but this makes sure that the plugin's instance URL matches the integration's base URL, essentially making sure it's the same Jira instance. It's possible to install the Jira integration multiple times for multiple instances, and it's also possible to install Jira _and_ Jira Server, so this checks that the right issues are migrated.